src/bp/sdk/botpress.d.ts

```diff
@@ -2136,6 +2147,24 @@ declare module 'botpress/sdk' {
      * @returns boolean indicating if code was valid & enough usage were left
      */
     export function consumeInviteCode(workspaceId: string, inviteCode?: string): Promise<boolean>
+
+    /**
+     * Retreives users in a given workspace. A user's object will be hydrated with desired attributes
+     * @param workspaceId
+     * @param attributes list of desired attributes
+     * @returns all users in workspace with desired attributes
+     */
+    export function getWorkspaceUsersWithAttributes(
+      workspaceId: string,
+      attributes?: string[]
+    ): Promise<WorkspaceUserWithAttributes[]>
+
+    /**
+     * Retreives users in a given workspace
+     * @param workspaceId
+     * @returns all users in workspace only email and strategy
+     */
+    export function getWorkspaceUsers(workspaceId: string): Promise<WorkspaceUser[]>
```

Comment:
This is probably overkill but if you wanted the return type to have attributes defined when attributes are asked, you would do something like this:

```ts
getWorkspaceUsers<O extends Partial<Options> | undefined>(workspaceId: string, options: O): Promise<(
    O extends { attributes: string[] } ?
    WorkspaceUserWithAttributes[] 
    :  WorkspaceUser[]
    )>
```

Then again, I think its very fine if `WorkspaceUser` has an optionnal `attributes` field.